File: 2044. Count Number of Maximum Bitwise-OR Subsets.cs
public class Solution {
    public int CountMaxOrSubsets(int[] nums)
    {
        int max = 0;
        foreach (int num in nums)
        {
            max |= num;
        }
        int count = 0;
        void Backtrack(int index, int curr)
        {
            if (index == nums.Length)
            {
                if (curr == max)
                {
                    count++;
                }
                return;
            }
            Backtrack(index + 1, curr | nums[index]);
            Backtrack(index + 1, curr);
        }
        Backtrack(0, 0);
        return count;
    }
}

File: 2044. Count Number of Maximum Bitwise-OR Subsets.py
class Solution:
    def countMaxOrSubsets(self, nums: List[int]) -> int:
        max_or = reduce(ior, nums)
        count = 0

        for i in nums:
            max_or |= i
        return self._count_subsets(nums, 0, 0, max_or)

    def _count_subsets(self, nums: List[int], index: int, current_or: int, target_or: int) -> int:
        if index == len(nums):
            return 1 if current_or == target_or else 0
        
        count_without = self._count_subsets(nums, index + 1, current_or, target_or)

        count_with = self._count_subsets(nums, index + 1, current_or | nums[index], target_or)

        return count_without + count_with
